### PR TITLE
Web - Verify Email

### PR DESCRIPTION
**Pull request by Rikimbili**
Review after [Web - Improve Data Handling](https://github.com/poop-team/daydream/pull/137) is merged

**Web - Forgot Password** will be going after this one

### CODE REVIEW

**Review by kitorer — COMMENTED** [review]
I see another mega PR

File: web/src/pages/api/user/login.ts [hunk 1]
```diff
@@ -2,53 +2,57 @@ import { compare } from "bcrypt";
...
 }
 
 export default async function Login(req: Request, res: NextApiResponse) {
   if (!validateMethod("POST", req, res)) return;
 
-  const { email, password } = req.body;
+  const { email, name, password } = req.body;
 
-  if (!email.trim() || !password.trim()) {
+  if (!(email?.trim() || name?.trim()) || !password.trim()) {
```
Comment (rob-3) [inline web/src/pages/api/user/login.ts:22]: This reads better as `if (!email?.trim() || !name?.trim() || !password.trim())`

**Review by rob-3 — CHANGES_REQUESTED** [review]

File: web/src/pages/api/user/login.ts (hunk 1 above)
Comment (Rikimbili) [reply web/src/pages/api/user/login.ts:22]: I believe that wouldn't work in this case. I'm allowing the consumer to send either a name OR email. So if they send an email without a name it should proceed as both are unique and we can find the user with either. However, they should always send a password

Comment (Rikimbili) [reply web/src/pages/api/user/login.ts:22]: we can apply de morgan's and get `!email?.trim() && !name?.trim() || !password.trim()` but it is arguable which is more readable

Comment (rob-3) [reply web/src/pages/api/user/login.ts:22]: Ah I see. I can't do boolean logic.

**Review by rob-3 — APPROVED** [review]

**Pull request merged**